### PR TITLE
Added info() in Trainer

### LINKED ISSUE #120: Add `Trainer.info()`
When a user wants to train a model using the `Trainer`, it just starts with no further info about the training properties. It's better to provide a function in the `Trainer ` to generate a user-friendly info message containing all the necessary propreties of the trainer. 
The properties I have in mind right now are:
- Model type
- Device(s)
- Dataset
- Optimizer
- Initial learning rate
- Total samples
- Epochs
- Batch size
- Total parameters
- Trainable parameters
- Mixed precision
- Metrics
- Checkpoints path

The output is something like below:
```
******************** Training Info ********************

Model type: T5TextGeneration
Device(s): ['cuda']
Dataset: `hezarai/xlsum-fa` (text_summarization)
Optimizer: `adam`
Initial learning rate: ...
Total samples: ...
Epochs: ...
Batch size: ...
Total parameters: ...
Trainable parameters: ...
Mixed precision: True
Metrics: ['rouge']
Checkpoints path: ...

*****************************************************

```

To wrap up, the task is to add a new method to the `Trainer` class called `info()` that just prints the above properties. Next, this method should be called in `Trainer.train()` before everything.

### PR DESCRIPTION
**Pull request by SSBakh07**
# Pull Request

## Description
Added a method `info()` to print training properties before model training actually begins.

## Changes
- Added info() method in the Trainer class (trainer.py)
- Call `self.info()` in the `train` function in the Trainer class

## Related Issues
Resolves #120 

## Checklist

- [x] I have read and followed the project's contributing guidelines.
- [x] My code follows the project's coding style.
- [x] I have tested my changes thoroughly.
- [x] My changes do not introduce any new warnings or errors.

### CODE REVIEW

File: hezar/trainer/trainer.py [hunk 1]
```diff
@@ -129,6 +129,25 @@ def _set_seed(seed) -> None:
...
         random.seed(seed)
+    
+    def info(self) -> None:
+        print("******************** Training Info ********************")
+        print(f"Task: {self.config.task}")
+        print(f"Model type: {type(self.model).__name__}")
+        print(f"Device(s): {self.device}")
+        print(f"Training Dataset: {self.train_dataset.config.name}")
+        print(f"Eval Dataset: {self.eval_dataset.config.name}")
+        print(f"Optimizer: {self.config.optimizer}")
```
Comment (arxyzan) [inline hezar/trainer/trainer.py:140]: This better be 
```python
self.config.optimizer or self.default_optimizer
```
Since most of the time optimizer is not set by the user and the default one is used.

File: hezar/trainer/trainer.py [hunk 2]
```diff
@@ -129,6 +129,25 @@ def _set_seed(seed) -> None:
         torch.manual_seed(seed)
         np.random.seed(seed)
         random.seed(seed)
+    
+    def info(self) -> None:
+        print("******************** Training Info ********************")
+        print(f"Task: {self.config.task}")
+        print(f"Model type: {type(self.model).__name__}")
+        print(f"Device(s): {self.device}")
+        print(f"Training Dataset: {self.train_dataset.config.name}")
```
Comment (arxyzan) [inline hezar/trainer/trainer.py:138]: I think datasets might better show their path on the hub? Like
```python
f"Training Dataset: {self.train_dataset.config.path}"
f"Eval Dataset: {self.eval_dataset.config.path}"
```

**Comment by arxyzan** [comment]
@SSBakh07 Great job Sheeda jan, I'm reviewed the changes and commented some minor points. It'd be great if you could resolve them. Thanks.
